src/rebar_prv_deps.erl

```diff
@@ -24,16 +24,25 @@ init(State) ->
                     {short_desc, "List dependencies"},
                     {desc, "List dependencies. Those not matching lock files "
                            "are followed by an asterisk (*)."},
-                    {opts, []}])),
+                    {opts, [{tree, $t, "tree", undefined, "Display dependencies in tree format."}]}])),
     {ok, State1}.
 
 -spec do(rebar_state:t()) -> {ok, rebar_state:t()} | {error, string()}.
 do(State) ->
-    Profiles = rebar_state:current_profiles(State),
-    List = [{Profile, rebar_state:get(State, {deps, Profile}, [])}
-            || Profile <- Profiles],
-    [display(State, Profile, Deps) || {Profile, Deps} <- List],
-    {ok, State}.
+    case display_tree(State) of
+        true ->
+            {_Packages, Graph} = rebar_state:packages(State),
+            List = merge_deps_per_profile(State),
+            {_SrcDeps, PkgDeps} = rebar_prv_install_deps:parse_deps("", List, State, [], 0),
+            rebar_digraph:print_solution(Graph, PkgDeps),
```

Comment:
Am I wrong or the resulting tree doesn't respect the locks vs. asterisks and types being printed like the rest of the module?